src/Microsoft.AspNet.Mvc.Core/Rendering/Html/HtmlHelper.cs

```diff
@@ -971,16 +971,8 @@ protected virtual string GenerateValue(string expression, object value, string f
             }
             else if (useViewData)
             {
-                if (string.IsNullOrEmpty(expression))
-                {
-                    // case 2(a): format the value from ViewData for the current model
-                    resolvedValue = FormatValue(ViewData.Model, format);
-                }
-                else
-                {
-                    // case 2(b): format the value from ViewData
-                    resolvedValue = DefaultHtmlGenerator.EvalString(ViewContext, expression, format);
-                }
+                // case 2: format the value from ViewData
```

Comment:
yes, `EvalString()` uses `ViewData`.  note this comment matches the root of the removed 2(a) and 2(b).